docs/integration-testing.md

```diff
@@ -0,0 +1,70 @@
+# Mac Silicon (ARM) - Firefox
+docker run -d -it -p 4444:4444 -p 7900:7900 --shm-size 2g --add-host=bam-test-1.nr-local.net:host-gateway seleniarm/standalone-firefox:latest
+
+# Intel Silicon (x86) - Chrome
+docker run -d -it -p 4444:4444 -p 7900:7900 --shm-size 2g --add-host=bam-test-1.nr-local.net:host-gateway selenium/standalone-chromium:latest
+# Intel Silicon (x86) - Firefox
+docker run -d -it -p 4444:4444 -p 7900:7900 --shm-size 2g --add-host=bam-test-1.nr-local.net:host-gateway selenium/standalone-firefox:latest
+```
+
+**Links**: [x86](https://github.com/SeleniumHQ/docker-selenium) || [ARM](https://github.com/seleniumhq-community/docker-seleniarm)
+
+Once the local Selenium instance is running, you can execute `npm run wdio -- --selenium -b 'chrome'` or `npm run wdio -- --selenium -b 'firefox'` to start running the tests. If you would like to watch the browser as the tests execute, access the Selenium VNC server [here](http://localhost:7900/?autoconnect=1&resize=scale&password=secret).
+
+To run the tests against a remote Selenium instance, such as a Selenium Grid service, use the `--selenium-host <host>` and `--selenium-port <port>` arguments.
+
+## WDIO CLI
+
+The project wraps the WDIO execution in a custom CLI. When executing WDIO, you are not running the CLI that ships with WDIO. The customer CLI arguments are defined [here](../tools/wdio/args.mjs). The custom CLI will take arguments, construct a WDIO configuration, and execute WDIO.
+
+## Writing Tests
+
+All tests should exist in the path `tests/specs` for WDIO to pick them up. Tests should be grouped by the functionality being tested or scenario being ran. WDIO will launch a new browser for each file, execute all tests of that file in the browser, and close the browser. This means more files will launch more browsers improving test isolation but also increasing the amount of time it takes to run the full test suite. Keep this in mind when writing integration tests.
+
+- **DO** create a global `describe` wrapper to hold all the tests. The name of this wrapper will be used to name the test result in SauceLabs making it easier to review test failures.
+- **DO** create a `describe` to group tests that have similar pre/post test instructions.
+- **DON'T** save state between individual test cases.
+- **DO** load a new page to start each test.
+- **DO** use `beforeEach` to instantiate variables and state before each test.
+- **DO** use `afterEach` to cleanup global state after each test.
+- **DON'T** use `beforeAll` or `afterAll`.
+- **DO** use the `it` keyword to declare a test case that has no browser limitation.
+- **DO** use the `test()` function to declare a test case that has browser limitations.
+- **DO** avoid promise chaining and use `async/await` to test asynchronous code.
+- **DO** create custom commands in `tools/wdio/plugins/jil-commands.mjs` instead of writing utility methods for common testing functionality.
```

Comment:
Might should be renamed `custom-commands.mjs`?

Reply:
Agreed. I think I want to remove all references to `JIL` at some point.